src/main/java/com/tw/go/plugin/BuildStatusNotifierPlugin.java

```diff
@@ -102,6 +106,9 @@ private GoPluginApiResponse handleGetPluginSettingsConfiguration() {
         response.put(PLUGIN_SETTINGS_USERNAME, createField("Username", null, true, false, "2"));
         response.put(PLUGIN_SETTINGS_PASSWORD, createField("Password", null, true, true, "3"));
         response.put(PLUGIN_SETTINGS_OAUTH_TOKEN, createField("OAuth Token", null, true, true, "4"));
+        if (provider.pluginId().equals(GerritProvider.PLUGIN_ID)) {
```

Comment:
You're right, it would be much cleaner if the providers would contain the configurations especially if the template name is moved to there too. And if in the future there are more general or plugin specific configurations added that method would become really messy. I might as well make the improvement already at this point.